src/DotLiquid.Tests/Util/ObjectExtensionMethodsTests.cs

```diff
@@ -1,122 +1,163 @@
-        {
-            var page = new
-            {
-                category = "" // exists, but empty
-            };
-
-            Helper.AssertTemplateResult(
-            expected: @"
-  <h1></h1>
-",
-            template: @"{% if page.category %}
-  <h1>{{ page.category }}</h1>
-{% endif %}",
-            localVariables: Hash.FromAnonymousObject(new { page }));
-        }
-
-        [Test]
-        public void TestGetPropertyValue()
-        {
-            var keyValuePair = new KeyValuePair<string, object>("*key*", "*value*");
-
-            Assert.That(ObjectExtensionMethods.GetPropertyValue(null, "any"), Is.Null);
-            Assert.That(keyValuePair.GetPropertyValue("NonExistant"), Is.Null);
-            Assert.That(new KeyValuePair<string, object>("*key*", null).GetPropertyValue("Value"), Is.Null);
-
-            Assert.That(keyValuePair.GetPropertyValue("Key"), Is.EqualTo("*key*"));
-            Assert.That(keyValuePair.GetPropertyValue("Value"), Is.EqualTo("*value*"));
-        }
-    }
+using System;
+using System.Collections.Generic;
+using DotLiquid.Util;
+using NUnit.Framework;
+
+namespace DotLiquid.Tests.Util
+{
+    [TestFixture]
+    public class ObjectExtensionMethodsTests
+    {
+        private static readonly object NIL = null;
+
+        [Test]
+        public void TestSafeTypeInsensitiveEqual()
+        {
+            // Not equal
+            Assert.That(NIL.SafeTypeInsensitiveEqual("nil"), Is.False);
+            Assert.That("nil".SafeTypeInsensitiveEqual(null), Is.False);
+            Assert.That("a string".SafeTypeInsensitiveEqual("A STRING"), Is.False); // different case string equality
+
+            // Equals
+            Assert.That(NIL.SafeTypeInsensitiveEqual(null), Is.True); // null equalilty
+            Assert.That("a string".SafeTypeInsensitiveEqual("a string"), Is.True); // same type equality
+            Assert.That(1.SafeTypeInsensitiveEqual("1"), Is.True); // int to string equality
+            Assert.That(Int64.Parse("99").SafeTypeInsensitiveEqual(Int32.Parse("99")), Is.True); // long to int equality
+            Assert.That(2.0f.SafeTypeInsensitiveEqual("2.0"), Is.True);  // float to string equality
+            Assert.That(2.0d.SafeTypeInsensitiveEqual("2.0"), Is.True);  // double to string equality
+        }
+
+        [Test]
+        public void TestSafeTypeInsensitiveEqualArrays()
+        {
+            string[] array1 = { "a", "b", "c" };
+            string[] array2 = { "a", "b", "c" };
+            string[] array3 = { "a", "b", "d" };
+            string[] array4 = { "A", "B", "C" };
+            string[] array5 = { "b", "c", "a" };
+            string[] array6 = { "a", "b", "c", "d" };
+            string[] array7 = { };
+            string[] array8 = { "1", "2", "3" };
+            List<string> list1 = new List<string> { "a", "b", "c" };
+            int[] intArray1 = { 1, 2, 3 };
+            int[] intArray2 = { 1, 2, 3 };
+            int[] intArray3 = { 1, 2, 3, 4, 5 };

```

Comment:
PR is good, but this test is not readable. Thank you AI for the suggestions.
```suggestion
            string[] alphabetSubset1 = { "a", "b", "c" };
            string[] alphabetSubset2 = { "a", "b", "c" };
            string[] alphabetSubsetDifferent = { "a", "b", "d" };
            string[] uppercaseAlphabetSubset = { "A", "B", "C" };
            string[] shuffledAlphabetSubset = { "b", "c", "a" };
            string[] extendedAlphabetSubset = { "a", "b", "c", "d" };
            string[] emptyArray = { };
            string[] numberStrings = { "1", "2", "3" };
            List<string> alphabetSubsetList = new List<string> { "a", "b", "c" };

            int[] numberArray1 = { 1, 2, 3 };
            int[] numberArray2 = { 1, 2, 3 };
            int[] extendedNumberArray = { 1, 2, 3, 4, 5 };
```